segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3StorageTest.java

```diff
@@ -0,0 +1,265 @@
+        BlobStore blobStore = context.getBlobStore();
+
+        s3Proxy = S3Proxy.builder().awsAuthentication( AuthenticationType.AWS_V2_OR_V4, "x", "x")
+                .endpoint(uri)
+                .keyStore("", "")
+                .blobStore(blobStore)
+                .ignoreUnknownHeaders(true)
+                .build();
+
+        s3Proxy.start();
+
+        this.adapterConfig = ExtendedS3StorageConfig.builder()
+                                                    .with(ExtendedS3StorageConfig.BUCKET, "kanpravegatest")
+                                                    .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                                    .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                                    .with(ExtendedS3StorageConfig.ROOT, "test")
+                                                    .with(ExtendedS3StorageConfig.URI, endpoint)
+                                                    .build();
+        if (client == null) {
+            createStorage();
+
+            try {
+                client.createBucket("kanpravegatest");
+            } catch (S3Exception e) {
+                if (!e.getErrorCode().equals("BucketAlreadyOwnedByYou")) {
+                    throw e;
+                }
+            }
+            List<ObjectKey> keys = client.listObjects("kanpravegatest").getObjects().stream().map((object) -> {
+                return new ObjectKey(object.getKey());
+            }).collect(Collectors.toList());
+
+            if (!keys.isEmpty()) {
+                client.deleteObjects(new DeleteObjectsRequest("kanpravegatest").withKeys(keys));
+            }
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+            client.shutdown();
+            client = null;
+            s3Proxy.stop();
+    }
+
+    @Override
+    protected Storage createStorage() {
+        S3Config s3Config = null;
+        URI uri = URI.create(endpoint);
+        s3Config = new S3Config(uri);
+        s3Config.withIdentity(adapterConfig.getAccessKey()).withSecretKey(adapterConfig.getSecretKey());
+
+        client = new S3JerseyClientWrapper(s3Config, aclMap);
+
+        ExtendedS3Storage storage = new ExtendedS3Storage(client, adapterConfig, executorService());
+        return storage;
+    }
+
+    @Override
+    protected SegmentHandle createHandle(String segmentName, boolean readOnly, long epoch) {
+        FileChannel channel = null;
+        if (readOnly) {
+            return ExtendedS3SegmentHandle.getReadHandle(segmentName);
+        } else {
+            return ExtendedS3SegmentHandle.getWriteHandle(segmentName);
+        }
+    }
+
+    /**
+     * Wrapper over S3JerseyClient. This implements ACLs, multipart copy and multiple writes to the same object on top of S3Proxy implementation.
+     */
+    private static class S3JerseyClientWrapper extends S3JerseyClient {
+        private final ConcurrentMap<String, AclSize> aclMap;
```

Comment:
I'm confused, don't we have an `S3ClientWrapper`extending `S3JerseyClient` already?